fhir-swagger-generator/src/main/java/com/ibm/fhir/openapi/generator/FHIROpenApiGenerator.java

```diff
@@ -63,112 +116,234 @@
+                            "Resource".equals(parentClassName) ||
+                            "Bundle".equals(parentClassName) ||
+                            "OperationOutcome".equals(parentClassName)) {
+                        Class<?> innerModelClass = Class.forName(RESOURCEPACKAGENAME + "." + innerClassName);
+                        // System.out.println("Resource:  " + className);
+                        generateDefinition(innerModelClass, definitions);
+                    }
+                }
+
+                swagger.add("tags", tags);
+                swagger.add("paths", paths);
+        
+                JsonObjectBuilder components = factory.createObjectBuilder();
+                JsonObjectBuilder parameters = factory.createObjectBuilder();
+                generateSearchParameters(parameters, filter);
+                JsonObject parametersObject = parameters.build();
+                if (!parametersObject.isEmpty()) {
+                    components.add("parameters", parametersObject);
+                }
+                components.add("requestBodies", requestBodies);
+                components.add("schemas", definitions);
+        
+                swagger.add("components", components);
+        
+                Map<String, Object> config = new HashMap<String, Object>();
+                config.put(JsonGenerator.PRETTY_PRINTING, true);
+                JsonWriterFactory factory = Json.createWriterFactory(config);
+                
+                File outFile = new File("./src/main/resources/openapi/" + resourceClassName + "-openapi.json");
+                try (JsonWriter writer = factory.createWriter(new FileWriter(outFile))) {
+                    writer.writeObject(swagger.build());
+                } catch (Exception e) {
+                    throw new Error(e);
+                }
+            }
+        }
+        
+        generateMetadataOpenApi();
+        generateBatchTransactionOpenApi();
+    }
+    
+    
+
+    private static void generateMetadataOpenApi() throws Exception, ClassNotFoundException, Error {
+        JsonObjectBuilder swagger = factory.createObjectBuilder();
+        swagger.add("swagger", "2.0");
+        
         JsonObjectBuilder info = factory.createObjectBuilder();
-        info.add("title", "FHIR REST API");
-        info.add("description", "IBM FHIR Server API");
+        info.add("title", "Capabilities");
+        info.add("description", "The capabilities interaction retrieves the information about a server's capabilities - which portions of the HL7 FHIR specification it supports.");
         info.add("version", "4.0.0");
         swagger.add("info", info);
 
-        /**
-         * "servers": [ { "url": "/fhir-server/api/v4" } ]
-         */
-        JsonArrayBuilder servers = factory.createArrayBuilder();
-        JsonObjectBuilder server = factory.createObjectBuilder();
-        server.add("url", "/fhir-server/api/v4");
```

Comment:
should we hard coded or get from server.xml?  I think maybe we should allow to configure/change rootcontext in server.xml.

Reply:
its still hard-coded, but i moved it to a constant at the top of the file and added a comment